fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/typing/Type.java

```diff
@@ -2154,7 +2155,7 @@ private static List<Field> normalizeFields(@Nullable final List<Field> fields) {
                     fieldToBeAdded =
                             new Field(field.getFieldType(),
                                     Optional.of(explicitFieldName),
-                                    Optional.of(field.getFieldIndexOptional().orElse(i + 1)));
+                                    Optional.of(i + 1));
```

Comment:
This is the only substantive code change. If we're in this part of the loop (as one of the comments states), the field names and protobuf indexes have been identified as fair game to change. We could update this to make a more concerted effort to change this index numbers more (e.g., by giving it a new index if it is one of the duplicate values), but it shouldn't matter

Reply:
That makes sense to me.